lib/ansible/plugins/inventory/__init__.py

```diff
@@ -292,21 +293,28 @@ def _add_host_to_keyed_groups(self, keys, variables, host, strict=False):
             for keyed in keys:
                 if keyed and isinstance(keyed, dict):
                     prefix = keyed.get('prefix', '')
+                    if prefix:
+                        prefix = prefix + "_"
```

Comment:
This does not allow for prefixes w/o _, if the user wants to add/ignore _ he should be able to

Reply:
In the previous code, if you didn't specify a prefix your group names would just start with an underscore, so I fail to see how this is worse.

Reply:
nvmd, missed the composition below

Reply:
but you actually remove teh prefix?